java/kotlin-extractor2/src/main/kotlin/KotlinExtractor.kt

```diff
@@ -166,6 +166,7 @@ OLD: KE1
         )
         dtw.flush()
         loggerBase.close()
+        System.exit(0) // TODO: figure out what's keeping the JVM awake
```

Comment:
Filed a ticket for this here: https://youtrack.jetbrains.com/issue/KT-73753/Analysis-API-doesnt-terminate

Reply:
You might want to include the context at https://github.com/github/codeql/pull/18215#issuecomment-2520120125 identifying thread-pool cleanup as the cause